### PR TITLE
Add exp flag for self managed cluster API upgrades

### PR DESCRIPTION
**Pull request by g-gaston**
## Description of changes

This allows us to test management cluster CP upgrades through the controller.

Not intended for the final user, only for internal testing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: controllers/cluster_controller.go [hunk 1]
```diff
@@ -234,7 +252,7 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, log logr.Logger, clus
 		return reconcileResult.ToCtrlResult(), nil
 	}
 
-	if cluster.IsSelfManaged() {
+	if cluster.IsSelfManaged() && !r.experimentalSelfManagedUpgrade {
```
Comment (ddjjia) [inline controllers/cluster_controller.go:255]: Are you going to still use the existing interface `ReconcileWorkerNodes` when the flag set to true to reconcile worker node?

Comment (g-gaston) [reply controllers/cluster_controller.go:258]: if the flag is set to true, we just call `Reconcile`, which will also reconcile worker nodes aside from CP :)

File: controllers/cluster_controller.go [hunk 2]
```diff
@@ -47,6 +47,10 @@ type ClusterReconciler struct {
 	awsIamAuth                 AWSIamConfigReconciler
 	clusterValidator           ClusterValidator
 	packagesClient             PackagesClient
+
+	// experimentalSelfManagedUpgrade enables management cluster full upgrades (including CP)
+	// Only intended for internal testing.
```
[the remark this answers is not in the document]
Comment (g-gaston) [reply controllers/cluster_controller.go]: done

[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply controllers/cluster_controller.go]: Awesome, thanks!

**Comment by g-gaston** [comment]
Merging manually, codecov only hits 60% but the only part not tested is the feature flag definition, which doesn't need a test.

**Pull request merged**